File: frontend/src/models/task.ts
export type TaskStatus = "In Progress" | "In Review" | "Completed";

export interface Task {
  id: string;

  clientId: string;

  accountantId: string;

  title: string;

  task_description: string;

  task_status: TaskStatus;

  created_at: string;

  updated_at: string | null;

  deleted_at: string | null;
}

File: backend/src/main/java/com/csci/tax_docs_portal/entity/Task.java
package com.csci.tax_docs_portal.entity;

import java.time.LocalDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

  private UUID id;

  private UUID clientId;

  private UUID accountantId;

  private String title;

  private String description;

  private String taskStatus;

  private LocalDateTime createdAt;

  private LocalDateTime updatedAt;

  private LocalDateTime deletedAt;
}
